### ISSUE TITLE
AssertionError: The optimization step for ELBO update returned a NaN while running DetermineGermlineContigPloidy

### ISSUE THREAD

**Issue by NTNguyen13** [broadinstitute/gatk#6573]
----

## Bug Report
Hi, I'm trying the CNV detection pipeline from GATK: https://gatk.broadinstitute.org/hc/en-us/categories/360002310591
However, when running the Determine Germline Contig Ploidy step, I stumble upon this error. Please guide me to solve this problem

### Affected tool(s) or class(es)
```
gatk DetermineGermlineContigPloidy \
    -L /home/nguyen/RB1/RB1.cohort.gc.filtered.interval_list \
    --interval-merging-rule OVERLAPPING_ONLY \
    -I ... (63 tsv files output from CollectReadCounts)
```

### Affected version(s)
- GATK 4.1.6.1
### Description 
Full error log:
```
Traceback (most recent call last):
  File "/tmp/cohort_determine_ploidy_and_depth.380621677219090732.py", line 119, in <module>
    ploidy_task.engage()
  File "/home/nguyen/anaconda3/envs/gatk/lib/python3.6/site-packages/gcnvkernel/tasks/inference_task_base.py", line 339, in engage
    converged_continuous = self._update_continuous_posteriors()
  File "/home/nguyen/anaconda3/envs/gatk/lib/python3.6/site-packages/gcnvkernel/tasks/inference_task_base.py", line 395, in _update_continuous_posteriors
    assert not np.isnan(loss), "The optimization step for ELBO update returned a NaN"
AssertionError: The optimization step for ELBO update returned a NaN
11:09:59.446 DEBUG ScriptExecutor - Result: 1
11:09:59.447 INFO  DetermineGermlineContigPloidy - Shutting down engine
[April 28, 2020 11:09:59 AM ICT] org.broadinstitute.hellbender.tools.copynumber.DetermineGermlineContigPloidy done. Elapsed time: 0.17 minutes.
Runtime.totalMemory()=623902720
org.broadinstitute.hellbender.utils.python.PythonScriptExecutorException: 
python exited with 1
Command Line: python /tmp/cohort_determine_ploidy_and_depth.380621677219090732.py --sample_coverage_metadata=/tmp/samples-by-coverage-per-contig8606344533091962323.tsv --output_calls_path=/home/nguyen/Exec/gatk-4.1.6.0/ploidy-calls --mapping_error_rate=1.000000e-02 --psi_s_scale=1.000000e-04 --mean_bias_sd=1.000000e-02 --psi_j_scale=1.000000e-03 --learning_rate=5.000000e-02 --adamax_beta1=9.000000e-01 --adamax_beta2=9.990000e-01 --log_emission_samples_per_round=2000 --log_emission_sampling_rounds=100 --log_emission_sampling_median_rel_error=5.000000e-04 --max_advi_iter_first_epoch=1000 --max_advi_iter_subsequent_epochs=1000 --min_training_epochs=20 --max_training_epochs=100 --initial_temperature=2.000000e+00 --num_thermal_advi_iters=5000 --convergence_snr_averaging_window=5000 --convergence_snr_trigger_threshold=1.000000e-01 --convergence_snr_countdown_window=10 --max_calling_iters=1 --caller_update_convergence_threshold=1.000000e-03 --caller_internal_admixing_rate=7.500000e-01 --caller_external_admixing_rate=7.500000e-01 --disable_caller=false --disable_sampler=false --disable_annealing=false --interval_list=/tmp/intervals9095034526503666073.tsv --contig_ploidy_prior_table=/home/nguyen/RB1/RB1_contig_ploidy_priors.tsv --output_model_path=/home/nguyen/Exec/gatk-4.1.6.0/ploidy-model
        at org.broadinstitute.hellbender.utils.python.PythonExecutorBase.getScriptException(PythonExecutorBase.java:75)
        at org.broadinstitute.hellbender.utils.runtime.ScriptExecutor.executeCuratedArgs(ScriptExecutor.java:130)
        at org.broadinstitute.hellbender.utils.python.PythonScriptExecutor.executeArgs(PythonScriptExecutor.java:170)
        at org.broadinstitute.hellbender.utils.python.PythonScriptExecutor.executeScript(PythonScriptExecutor.java:151)
        at org.broadinstitute.hellbender.utils.python.PythonScriptExecutor.executeScript(PythonScriptExecutor.java:121)
        at org.broadinstitute.hellbender.tools.copynumber.DetermineGermlineContigPloidy.executeDeterminePloidyAndDepthPythonScript(DetermineGermlineContigPloidy.java:406)
        at org.broadinstitute.hellbender.tools.copynumber.DetermineGermlineContigPloidy.doWork(DetermineGermlineContigPloidy.java:303)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:139)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:191)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:210)
        at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:163)
        at org.broadinstitute.hellbender.Main.mainEntry(Main.java:206)
        at org.broadinstitute.hellbender.Main.main(Main.java:292)

```

Any help is warmly welcome, thank you

**Comment by ldgauthier** [comment]
@asmirnov239 you've looked into something like this before, right?

**Comment by asmirnov239** [comment]
@NTNguyen13 Hi thank you for the bug report, 

Could you see if your count files have chromosomes that have zero/low coverage? Did you use the FilterIntervals prior to using DetermineContigPloidy workflow?
 
Could you provide the standard output as well?

**Comment by asmirnov239** [comment]
Also, could you provide the sample x per-contig-count matrix output by DetermineGermlineContigPloidy? It is a temporary output - it's easier to find if you provide `--tmp-dir /path/to/temp/dir` option in the tool command line invocation.

**Comment by NTNguyen13** [comment]
Yes, I have used `FilterIntervals` prior to using `DetermineContigPloidy` to create the `RB1.cohort.gc.filtered.interval_list` file. Here are the interval list file and the sample x coverage matrix
[RB1.cohort.gc.filtered.interval_list.txt](https://github.com/broadinstitute/gatk/files/4562241/RB1.cohort.gc.filtered.interval_list.txt)
[samples-by-coverage-per-contig6967710280691192984.xlsx](https://github.com/broadinstitute/gatk/files/4562279/samples-by-coverage-per-contig6967710280691192984.xlsx)

**Comment by NTNguyen13** [comment]
@asmirnov239 hi, is there any solution to this problem yet? Thank you

**Comment by asmirnov239** [comment]
@NTNguyen13 Yes, I just merged the PR that fixes it https://github.com/broadinstitute/gatk/pull/6613 so it will be available in the next release. Thank you again for bringing this up.

**Comment by asmirnov239** [comment]
Closed by  #6613
